### PR TITLE
[WIP] Fix sockets timeout issue

### PR DESCRIPTION
**Pull request by ZIMkaRU**

### CODE REVIEW

File: workers/api.framework.report.wrk.js [hunk 1]
```diff
@@ -100,10 +108,19 @@ class WrkReportFrameWorkApi extends WrkReportServiceApi {
...
-    const conf = this.conf[this.group]
+    const workerPathAbsolute = path.join(
+      this.ctx.root,
+      'workers/loc.api/sync/dao/sqlite-worker/index.js'
+    )
+    const {
+      syncMode,
+      dbDriver,
+      verboseSql,
+      isSqliteStoredInMemory
```
Comment (ezewer) [inline workers/api.framework.report.wrk.js:119]: Note: isSqliteStoredInMemory is used for tests?

**Review by ezewer — COMMENTED** [review]
Wouldnt add the support for both options adds complexity and dont see any benefit on having both sqlite and better-sqlite at same time.

File: workers/api.framework.report.wrk.js (hunk 1 above)
Comment (ZIMkaRU) [reply workers/api.framework.report.wrk.js:119]: that is right
but right now for tests used a file to support WAL mode, without storing in memory

File: workers/loc.api/di/factories/migrations-factory.js [hunk 2]
```diff
@@ -8,9 +8,12 @@ const {
 } = require('../../errors')
 
 module.exports = (ctx) => {
-  const { dbDriver } = ctx.container.get(
+  const { dbDriver: _dbDriver } = ctx.container.get(
     TYPES.CONF
   )
+  const dbDriver = /sqlite/gi.test(_dbDriver)
```
Comment (ezewer) [inline workers/loc.api/di/factories/migrations-factory.js:14]: /sqlite/gi.test('better-sqlite') -> True
/sqlite/gi.test('sqlite') -> True

File: workers/loc.api/sync/authenticator/helpers/generate-sub-user-name.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+'use strict'
```
Comment (ezewer) [inline workers/loc.api/sync/authenticator/helpers/generate-sub-user-name.js:1]: This does not seem related to changing from sqlite to better-sqlite, is there a reason to add it here?

File: workers/loc.api/sync/authenticator/helpers/index.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+'use strict'
```
Comment (ezewer) [inline workers/loc.api/sync/authenticator/helpers/index.js:1]: Same not related to changing db

File: workers/loc.api/sync/authenticator/index.js [hunk 5]
```diff
@@ -199,7 +205,8 @@ class Authenticator {
         isDecryptedApiKeys: true,
         isFilledSubUsers: true,
         isReturnedPassword: true,
-        isNotInTrans
+        isNotInTrans,
```
Comment (ezewer) [inline workers/loc.api/sync/authenticator/index.js:208]: is not in trans, would mean is not in transaction right?

File: workers/loc.api/sync/authenticator/index.js [hunk 6]
```diff
@@ -199,7 +205,8 @@ class Authenticator {
...
+        withoutWorkerThreads
```
Comment (ezewer) [inline workers/loc.api/sync/authenticator/index.js:209]: Wouldnt let this to be configurable, would force users to work with threats if this was the point of adding better-qlite

File: workers/loc.api/sync/balance.history/index.js [hunk 7]
```diff
@@ -191,21 +203,30 @@ class BalanceHistory {
...
       ) => {
+        const accum = await promise
+
         const _isForexSymb = isForexSymb(currency, this.FOREX_SYMBS)
-        const price = _isForexSymb
-          ? null
-          : this._getCandlesClosePrice(
+        let price = null
+
+        if ((i % 100) === 0) {
```
Comment (ezewer) [inline workers/loc.api/sync/balance.history/index.js:216]: Not being able to follow the logic with the I here

File: workers/loc.api/sync/balance.history/index.js [hunk 8]
```diff
@@ -256,20 +277,27 @@ class BalanceHistory {
...
 
-    const resInUsd = dataArr.reduce((accum, [symb, balance]) => {
+    const resInUsd = await dataArr.reduce(async (
+      promise,
+      [symb, balance],
+      i
+    ) => {
+      const accum = await promise
+
+      if ((i % 10) === 0) {
```
Comment (ezewer) [inline workers/loc.api/sync/balance.history/index.js:287]: Same here with the 10

File: workers/loc.api/sync/balance.history/index.js [hunk 9]
```diff
@@ -256,20 +277,27 @@ class BalanceHistory {
...
-      const prise = {
-        ...this._getCandlesClosePrice(
-          candles,
-          mts,
-          timeframe,
-          `t${symb}USD`,
-          currenciesSynonymous
-        )
-      }
+      const prise = this._getCandlesClosePrice(
```
Comment (ezewer) [inline workers/loc.api/sync/balance.history/index.js:294]: typo here: is price. Please check where it repeats.

File: workers/loc.api/di/factories/db-migrator-factory.js [hunk 10]
```diff
@@ -12,7 +12,7 @@ module.exports = (ctx) => {
       TYPES.DAO
     )
 
-    if (dbDriver === 'sqlite') {
+    if (/sqlite/gi.test(dbDriver)) {
```
Comment (ezewer) [inline workers/loc.api/di/factories/db-migrator-factory.js:15]: /sqlite/gi.test('better-sqlite') -> True
/sqlite/gi.test('sqlite') -> True

File: workers/loc.api/sync/colls.accessors/public.colls.conf.accessors.js [hunk 11]
```diff
@@ -1,5 +1,7 @@
 'use strict'
 
+const { promisify } = require('util')
+const setImmediatePromise = promisify(setImmediate)
```
Comment (ezewer) [inline workers/loc.api/sync/colls.accessors/public.colls.conf.accessors.js:4]: Whats the logic behind adding setImmediatePromise now and then on this code?

File: workers/loc.api/sync/currency.converter/index.js [hunk 12]
```diff
@@ -165,6 +182,7 @@ class CurrencyConverter {
     }
 
     for (const [symbol, conversion] of synonymous) {
+      await setImmediatePromise()
```
Comment (ezewer) [inline workers/loc.api/sync/currency.converter/index.js:185]: why add setImmediatePromise if next line 
 
> const price = await finderFn(symbol)


Would already add this to the event loop.

**Review by ezewer — COMMENTED** [review]
Adding some comments
